test/blackbox-tests/test-cases/actions/actions-stdxxx-on-success.t

```diff
@@ -0,0 +1,221 @@
+Test for --actions-stdxxx-on-success
+====================================
+
+  $ export BUILD_PATH_PREFIX_MAP="sh=$(which sh):$BUILD_PATH_PREFIX_MAP"
+
+  $ echo '(lang dune 3.0)' > dune-project
+
+  $ cat > dune <<EOF
+  > (rule
+  >  (alias default)
+  >  (action (system "echo 'Hello, world!'")))
+  > 
+  > (rule
+  >  (alias default)
+  >  (action (system "echo 'Something went wrong!' >&2")))
+  > EOF
+
+By default, stdout and stderr are always printed:
+
+  $ dune build
+            sh alias default
+  Hello, world!
+            sh alias default
+  Something went wrong!
+
+swallow tests
+-------------
+
+  $ dune clean
+  $ dune build --actions-stdout-on-success=swallow --actions-stderr-on-success=swallow
+
+must-be-be-empty tests
+----------------------
+
+  $ dune clean
+  $ dune build --actions-stdout-on-success=must-be-empty
+            sh alias default
+  Something went wrong!
+  File "dune", line 1, characters 0-65:
+  1 | (rule
+  2 |  (alias default)
+  3 |  (action (system "echo 'Hello, world!'")))
+            sh alias default (exit 0)
+  (cd _build/default && sh -c 'echo '\''Hello, world!'\''')
+  Hello, world!
+  [1]
+
+  $ dune clean
+  $ dune build --actions-stderr-on-success=must-be-empty
+            sh alias default
+  Hello, world!
+  File "dune", line 5, characters 0-77:
+  5 | (rule
+  6 |  (alias default)
+  7 |  (action (system "echo 'Something went wrong!' >&2")))
+            sh alias default (exit 0)
+  (cd _build/default && sh -c 'echo '\''Something went wrong!'\'' >&2')
+  Something went wrong!
+  [1]
+
+Incremental builds
+------------------
+
+When changing the settings, actions that should be re-executed are
+indeed re-executed:
+
+  $ dune clean
+  $ dune build \
+  >   --actions-stdout-on-success=swallow \
+  >   --actions-stderr-on-success=swallow
+
+  $ dune build
+            sh alias default
+  Hello, world!
+            sh alias default
+  Something went wrong!
+
+However, we currently re-execute too much. In particular, we
+re-execute actions whose outcome is not affected by the change:
+
+  $ cat > dune <<EOF
+  > (rule
+  >  (alias default)
+  >  (action (system "echo a.stdout; echo a.stderr >&2")))
+  > 
+  > (rule
+  >  (alias default)
+  >  (action (system "echo b.stderr >&2")))
+  > EOF
+
+  $ dune clean
+  $ dune build --actions-stdout-on-success=swallow
+            sh alias default
+  a.stderr
+            sh alias default
+  b.stderr
+
+Here, it isn't necessary to re-execute the second
+rule but we still do it:
```

Comment:
I wasn't sure what exactly you mean here. Here is my understanding:

Dune could have remembered that the action's `stdout` was empty last time and therefore it doesn't need to be rerun with the `swallow` setting. However, skipping such unnecessary executions would require some pretty ad hoc logic in the build system, so it doesn't seem worth it.

If the above matches your understanding, let's add this clarification to the test?

Reply:
It didn't need to re-execute it both because it succeeded and because it had no stdout output. I'm clarifying all this

Reply:
done